packages/fx-core/src/plugins/solution/fx-solution/v2/deploy.ts

```diff
@@ -52,6 +53,25 @@ export async function deploy(
     );
   }
 
+  const appStudioTokenJson = await tokenProvider.appStudioToken.getJsonObject();
```

Comment:
What if user didn't sign in here?

Reply:
In that case, a UserError will be thrown. Returning undefined only occurs on rare cases like networking failed.